### PR TITLE
Support searching in header scope

### PR DESCRIPTION
**Pull request by ngphi**
Allowing the validator to search in the header request ( headers as alias ).
cc @buunguyen

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ __Options__
 
 `opts` is an object specifying the fields and their corresponding validation rules.
 
-* Each key is a field name in the post data (e.g. 'name', 'user.name') with optional search scopes: `query`, `body` and `params`. Field name and scopes are separated by `:`. If no scope is specified, all scopes are searched.
+* Each key is a field name in the post data (e.g. 'name', 'user.name') with optional search scopes: `header`, `headers`, `query`, `body` and `params`. Field name and scopes are separated by `:`. If no scope is specified, all scopes are searched.
```
Comment (buunguyen) [inline README.md]: Maybe make it clear that header and headers are equivalent. Like: "with optional search scopes: `header` (or `headers`), `query`...

File: src/index.js [hunk 2]
```diff
@@ -72,8 +72,7 @@ export default function validate(rules) {
 
       const value = scope === 'params'
         ? getValueByPath(ctx.params, name)
-        : scope === 'query'
-          ? getValueByPath(ctx.request.query, name)
+        : ctx.request[`${scope}`] ? getValueByPath(ctx.request[`${scope}`], name)
```
Comment (buunguyen) [inline src/index.js]: ctx.request[scope]

File: src/index.js [hunk 3]
```diff
@@ -72,8 +72,7 @@ export default function validate(rules) {
...
           : (ctx.request.body && getValueByPath(ctx.request.body, name))
```
Comment (buunguyen) [inline src/index.js]: Is this necessary anymore? It seems to me the whole thing could be rewritten as:

```
const data = ctx[scope] || ctx.request[scope];
const value = getValueAtPath(data, path); // note I rename 'name' to 'path' and the method name to be clearer
```

**Review by buunguyen — CHANGES_REQUESTED** [review]

**Comment by ngphi** [comment]
Fixed PR. I totally agree with your feedbacks. Many thanks, @buunguyen

**Comment by buunguyen** [comment]
Looks good, thanks for working on this.

**Pull request merged**